### PR TITLE
Cerebral - Simulador now exposes devices as REST API

### PR DESCRIPTION
**Pull request by armandoblanco**
- Enabled the metrics of the devices as REST API to have the same data in the Webapp, Grafana, Cerebral and send to MQ.
- Two methods are available for displaying devices and their metrics
- Enabled swagger to document the REST API contract.

### CODE REVIEW

**Review by fcabrera23 — CHANGES_REQUESTED** [review]
@armandoblanco  what is "previous_version"?

File: agora/cerebral_api/src/app2.py [hunk 1]
```diff
@@ -0,0 +1,710 @@
+from flask import Flask, request, jsonify, render_template
```
Comment (fcabrera23) [inline agora/cerebral_api/src/app2.py:1]: Why this is called app2.py?

File: agora/cerebral_api/src/app2.py [hunk 2]
```diff
@@ -0,0 +1,710 @@
...
+from indexer import DocumentIndexer
+import onnxruntime_genai as og
+
+# Configure logging
+logging.basicConfig(level=logging.INFO,
+                   format='%(asctime)s - %(levelname)s - %(message)s')
+
+# Initialize Flask and SocketIO
+app = Flask(__name__)
+app.config['SECRET_KEY'] = 'your-secret-key'
```
Comment (fcabrera23) [inline agora/cerebral_api/src/app2.py:27]: What is this secret? shouldnt be an env variable?

File: agora/cerebral_api/src/app2.py [hunk 3]
```diff
@@ -0,0 +1,710 @@
...
+            'temperature': 0.7,
+            'top_p': 0.9,
+            'top_k': 50,
+            'do_sample': True,
+            'repetition_penalty': 1.1,
+            'min_length': 1
+        }
+
+        self.rag_template = """<|user|>
+Context information:
```
Comment (fcabrera23) [inline agora/cerebral_api/src/app2.py:74]: correct identation

File: agora/cerebral_api/src/app2.py [hunk 4]
```diff
@@ -0,0 +1,710 @@
...
+        if request.content_type != 'application/json':
+            raise BadRequest('Content-Type must be application/json')
+        
+        data = request.get_json(force=True)
+        print(data)
+        industry = data.get('industry')
+        if not industry:
+            api.abort(400, 'Industry parameter is required')
+
+        print(industry)
```
Comment (fcabrera23) [inline agora/cerebral_api/src/app2.py:260]: remove print

File: agora/cerebral_api/src/app2.py [hunk 5]
```diff
@@ -0,0 +1,710 @@
...
+    @api.doc(responses={200: 'Success', 400: 'Missing required parameters', 404: 'Invalid industry provided'})
+    @api.expect(industry_model)
+    def post(self):
+        print("/api/roles")
+        """Fetch roles based on provided industry"""
+        if request.content_type != 'application/json':
+            raise BadRequest('Content-Type must be application/json')
+        
+        data = request.get_json(force=True)
+        print(data)
```
Comment (fcabrera23) [inline agora/cerebral_api/src/app2.py:255]: remove print

File: agora/cerebral_api/src/app2.py [hunk 6]
```diff
@@ -0,0 +1,710 @@
...
+industry_model = ns.model('Industry', {
+    'industry': fields.String(required=True, description='Specify the industry to retrieve roles')
+})
+
+@ns.route('/api/get_roles' , methods=['POST'])
+class Roles(Resource):
+    @api.doc(responses={200: 'Success', 400: 'Missing required parameters', 404: 'Invalid industry provided'})
+    @api.expect(industry_model)
+    def post(self):
+        print("/api/roles")
```
Comment (fcabrera23) [inline agora/cerebral_api/src/app2.py:249]: remove print

File: agora/cerebral_api/src/app2.py [hunk 7]
```diff
@@ -0,0 +1,710 @@
...
+
+        print(industry)
+        
+        # Flatten the industries list to a dictionary for easier lookup
+        industries_dict = {k: v for d in industries for k, v in d.items()}
+
+        if industry not in industries_dict:
+            api.abort(404, 'Invalid industry provided')
+
+        print(industries_dict[industry])
```
Comment (fcabrera23) [inline agora/cerebral_api/src/app2.py:268]: remove print

File: agora/cerebral_api/src/app2.py [hunk 8]
```diff
@@ -0,0 +1,710 @@
...
+    @api.doc(responses={200: 'Success', 400: 'Missing required parameters', 404: 'Invalid industry or role provided'})
+    @api.expect(alert_model)
+    def post(self):
+        print("/api/get_proactive_alerts")
+        """Fetch proactive alerts based on provided industry and role"""
+        if request.content_type != 'application/json':
+            raise BadRequest('Content-Type must be application/json')
+        
+        data = request.get_json(force=True)
+        print(data)
```
Comment (fcabrera23) [inline agora/cerebral_api/src/app2.py:341]: remove print

File: agora/cerebral_api/src/app2.py [hunk 9]
```diff
@@ -0,0 +1,710 @@
...
+        role = data.get('role')
+        if not industry or not role:
+            api.abort(400, 'Industry and role parameters are required')
+
+        print(f"Industry: {industry}, Role: {role}")
+
+        if industry not in industries or role not in industries[industry]:
+            api.abort(404, 'Invalid industry or role provided')
+
+        print(f"Alerts for {industry} - {role}")
```
Comment (fcabrera23) [inline agora/cerebral_api/src/app2.py:352]: remove print

File: agora/cerebral_api/src/app2.py [hunk 10]
```diff
@@ -0,0 +1,710 @@
...
+            raise BadRequest('Content-Type must be application/json')
+        
+        data = request.get_json(force=True)
+        print(data)
+        industry = data.get('industry')
+        role = data.get('role')
+        if not industry or not role:
+            api.abort(400, 'Industry and role parameters are required')
+
+        print(f"Industry: {industry}, Role: {role}")
```
Comment (fcabrera23) [inline agora/cerebral_api/src/app2.py:347]: remove print

File: agora/cerebral_api/src/app2.py [hunk 11]
```diff
@@ -0,0 +1,710 @@
...
+@ns.route('/api/convert_question_query_sql')
+class ConvertQuestionQuerySQL(Resource):
+    @api.doc(responses={200: 'Success', 400: 'Missing required parameters'})
+    @api.expect(question_model)
+    def post(self):
+        """Converts question to SQL query"""
+        if request.content_type != 'application/json':
+            raise BadRequest('Content-Type must be application/json')
+        
+        #print(request)
```
Comment (fcabrera23) [inline agora/cerebral_api/src/app2.py:430]: remove print

File: agora/cerebral_api/src/indexer.py [hunk 12]
```diff
@@ -0,0 +1,198 @@
...
+from langchain.document_loaders import DirectoryLoader, TextLoader, PyPDFLoader
+from dotenv import load_dotenv
+import logging
+
+# Configure logging
+logging.basicConfig(level=logging.INFO,
+                   format='%(asctime)s - %(levelname)s - %(message)s')
+
+# Load environment variables
+load_dotenv()
```
Comment (fcabrera23) [inline agora/cerebral_api/src/indexer.py:17]: comment

File: agora/cerebral_simulator/src/app.py [hunk 13]
```diff
@@ -10,22 +10,18 @@
...
+from flasgger import Swagger, swag_from
 
 from store_simulator import run_store_simulator
 
 #dev mode
-#from dotenv import load_dotenv
-#load_dotenv()
-
+from dotenv import load_dotenv
+load_dotenv()
```
Comment (fcabrera23) [inline agora/cerebral_simulator/src/app.py:24]: comment

**Review by fcabrera23 — CHANGES_REQUESTED** [review]

File: agora/cerebral_api/src/app2.py (hunk 1 above)
Comment (armandoblanco) [reply agora/cerebral_api/src/app2.py]: I deleted this file

File: agora/cerebral_api/src/app2.py (hunk 2 above)
Comment (armandoblanco) [reply agora/cerebral_api/src/app2.py]: I deleted this file

File: agora/cerebral_api/src/app2.py (hunk 3 above)
Comment (armandoblanco) [reply agora/cerebral_api/src/app2.py]: I deleted this file

File: agora/cerebral_api/src/app2.py (hunk 4 above)
Comment (armandoblanco) [reply agora/cerebral_api/src/app2.py]: I deleted this file

File: agora/cerebral_api/src/app2.py (hunk 5 above)
Comment (armandoblanco) [reply agora/cerebral_api/src/app2.py]: I deleted this file

File: agora/cerebral_api/src/app2.py (hunk 9 above)
Comment (armandoblanco) [reply agora/cerebral_api/src/app2.py]: I deleted this file

File: agora/cerebral_api/src/app2.py (hunk 10 above)
Comment (armandoblanco) [reply agora/cerebral_api/src/app2.py]: I deleted this file

File: agora/cerebral_api/src/app2.py (hunk 11 above)
Comment (armandoblanco) [reply agora/cerebral_api/src/app2.py]: I deleted this file

File: agora/cerebral_simulator/src/app.py (hunk 13 above)
Comment (armandoblanco) [reply agora/cerebral_simulator/src/app.py]: I have updated this line

**Pull request merged**